src/main/java/org/broadinstitute/consent/http/resources/Resource.java

```diff
@@ -153,7 +153,9 @@ private static Response unableToExecuteExceptionHandler(Exception e) {
         try {
             if (e.getCause() instanceof PSQLException) {
                 String vendorCode = ((PSQLException) e.getCause()).getSQLState();
-                status = vendorCodeStatusMap.get(vendorCode);
+                if (vendorCodeStatusMap.containsKey(vendorCode)) {
+                    status = vendorCodeStatusMap.get(vendorCode);
+                }
```

Comment:
If we didn't have a vendor code in the map, this would reset the status to null which triggered an NPE in the returned response object.